# New-ish GCC versions in Ubuntu mistakenly detect buffer overflows in
# Python 2.3; imported osdefs.h from 2.4 that doesn't have this issue. See
# https://bugs.launchpad.net/ubuntu/+source/gcc-defaults/+bug/286334
# for a more in-depth description of this issue.

--- a/Include/osdefs.h
+++ b/Include/osdefs.h
@@ -7,13 +7,6 @@
 
 /* Operating system dependencies */
 
-#ifdef macintosh
-#define SEP ':'
-#define MAXPATHLEN 256
-/* Mod by Jack: newline is less likely to occur in filenames than space */
-#define DELIM '\n'
-#endif
-
 /* Mod by chrish: QNX has WATCOM, but isn't DOS */
 #if !defined(__QNX__)
 #if defined(MS_WINDOWS) || defined(__BORLANDC__) || defined(__WATCOMC__) || defined(__DJGPP__) || defined(PYOS_OS2)
@@ -44,8 +37,12 @@
 
 /* Max pathname length */
 #ifndef MAXPATHLEN
+#if defined(PATH_MAX) && PATH_MAX > 1024
+#define MAXPATHLEN PATH_MAX
+#else
 #define MAXPATHLEN 1024
 #endif
+#endif
 
 /* Search path entry delimiter */
 #ifndef DELIM
